File: Python/Problem_1758.py
# Problem - 1758: https://leetcode.com/problems/minimum-changes-to-make-alternating-binary-string/
# Lang: Python3
class Solution:
    def minOperations(self, s: str) -> int:
        operations = 0

        for i in range(len(s)):
            if i % 2 == 0:
                operations += 1 if s[i] != '0' else 0
            else:
                operations += 1 if s[i] != '1' else 0

        return min(operations, len(s) - operations)


if __name__=="__main__":
    sol = Solution()

    print(sol.minOperations("0100"))
    print(sol.minOperations("10"))
    print(sol.minOperations("1111"))

File: Cpp/Problem_1758.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
	int minOperations(string s) {
		int operations = 0;

		for (int i = 0; i < s.size(); i++) {
			if (i % 2 == 0)
				s[i] != '0' ? operations++ : true;
			else
				s[i] != '1' ? operations++ : true;
		}

		return min(operations, int(s.size()) - operations);
	}
};

int main() {
	Solution sol;

	cout << sol.minOperations("0100") << endl;
	cout << sol.minOperations("10") << endl;
	cout << sol.minOperations("1111") << endl;

	return 0;
}